src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2967,7 +2967,7 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
                 }
                 catch (Cci.PeWritingException e)
                 {
-                    diagnostics.Add(MessageProvider.CreateDiagnostic(MessageProvider.ERR_PeWritingFailure, Location.None, e.InnerException.ToString()));
+                    diagnostics.Add(MessageProvider.CreateDiagnostic(MessageProvider.ERR_PeWritingFailure, Location.None, e.InnerException?.ToString() ?? ""));
```

Comment:
📝 `MessageProvider.CreateDiagnostic` is incorrectly annotated. It should accept `params object?[]` like all other formatting methods.

Reply:
Possibly. We've gone back and forth on that particular API. The assumption from a lot of the team is that it was non-null and it seemed to take little work to make that true. But if we hit many more items like this as we move outward with our annotations I think we'll need to reconsider.